### PR TITLE
Homework 21

### PR DESCRIPTION
**Pull request by carlottaoh**

### CODE REVIEW

File: src/test/java/Homework21.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    public void renamePlaylist() {
+
+        String email = "demo@class.com";
+        String password = "te$t$tudent";
+
+        enterCredentials(email, password);
+        clickLogin();
+
+        editPlaylistName();
+        editPlaylistBack();
```
Comment (Berichicko) [inline src/test/java/Homework21.java]: **Use in test methods, the following logical split:   GIVEN – WHEN - THEN**
    
For example

```
 @Test

    public void testChangeProfileName (){
        // GIVEN

        String newName  = getRandomString();

        String currentPassword = "te$t$tudent";

        provideEmail("demo@class.com");

        providePassword();

        clickSubmitBtn();

 

        // WHEN

        openUserProfilePage();

        setName(newName);

        setCurrentPassword(currentPassword);

 

 

        // THEN

        Assert.assertTrue(getSuccessPopUp().isDisplayed());

}
```

File: src/test/java/BaseTest.java [hunk 2]
```diff
@@ -96,4 +101,32 @@ public void addSongNewList(String playlistName) {
         enterElement.click();
     }
 
+    public void editPlaylistName() {
+
+        WebElement findExisting = driver.findElement(By.xpath(".//a[contains(text(),\"CC's new playlist\")]"));
```
Comment (Berichicko) [inline src/test/java/BaseTest.java:106]: Please finish the homework!You got this problem, please correct it. 


image




HINT

Hence, press Control+F. A text field will appear. Write the expression and check if it matches with an element in the webpage. If it is matching, then copy the expression and paste in selenium code so we can avoid NoSuchElemetException

image

**Comment by carlottaoh** [comment]
It looks like someone deleted the playlist titled "CC's new playlist". The test is passing for me if the list is there to start with.

**Comment by Berichicko** [comment]
> It looks like someone deleted the playlist titled "CC's new playlist". The test is passing for me if the list is there to start with.

Please finish the homework!You got this problem, please correct it. 

image

File: src/test/java/Homework21.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        WebElement successMsg = driver.findElement(By.cssSelector(".success.show"));
+        WebElement successMsgVisible = wait.until(ExpectedConditions.visibilityOf(successMsg));
+        Assert.assertTrue(successMsgVisible.isDisplayed());
+
+        editPlaylistBack();
+        WebElement newName = driver.findElement(By.xpath(".//a[contains(text(),\"CC's new playlist\")]"));
+        Assert.assertTrue(newName.isDisplayed());
+        WebElement successMsg2 = driver.findElement(By.cssSelector(".success.show"));
+        WebElement successMsgVisible2 = wait.until(ExpectedConditions.visibilityOf(successMsg2));
+        Assert.assertTrue(successMsgVisible2.isDisplayed());
```
Comment (Berichicko) [inline src/test/java/Homework21.java:31]: **Use in test methods, the following logical split:   GIVEN – WHEN - THEN**
    
For example

```
 @Test

    public void testChangeProfileName (){
        // GIVEN

        String newName  = getRandomString();

        String currentPassword = "te$t$tudent";

        provideEmail("demo@class.com");

        providePassword();

        clickSubmitBtn();

 

        // WHEN

        openUserProfilePage();

        setName(newName);

        setCurrentPassword(currentPassword);

 

 

        // THEN

        Assert.assertTrue(getSuccessPopUp().isDisplayed());

}
```


**One test - One Assert**

**Pull request merged**